File: ios/Core/UI/Sources/CoreUI/Preview/MultiplayerPreview.swift
import SwiftUI

public struct MultiplayerPreview<Content: View>: View {
    private let colorScheme: ColorScheme
    private let alignment: Alignment
    private let content: Content

    public init(
        colorScheme: ColorScheme = .dark,
        alignment: Alignment = .topLeading,
        @ViewBuilder content: () -> Content
    ) {
        self.colorScheme = colorScheme
        self.alignment = alignment
        self.content = content()
    }

    public var body: some View {
        MultiplayerDesignSystem(colorScheme: colorScheme) {
            MultiplayerPreviewContainer(
                alignment: alignment,
                content: { content }
            )
        }
    }
}

private struct MultiplayerPreviewContainer<Content: View>: View {
    @Environment(\.multiplayerTheme) private var theme

    let alignment: Alignment
    let content: () -> Content

    var body: some View {
        ZStack(alignment: alignment) {
            MultiplayerBrandBackground()

            content()
                .padding(theme.spacing.lg)
                .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: alignment)
        }
    }
}

#Preview("Multiplayer Preview") {
    MultiplayerPreview(alignment: .center) {
        PreviewCard()
    }
}

private struct PreviewCard: View {
    @Environment(\.multiplayerTheme) private var theme

    var body: some View {
        MultiplayerSurface(
            cornerRadius: theme.radius.large,
            color: theme.colors.surfacePrimary,
            elevation: theme.elevation.level2
        ) {
            MultiplayerText(
                verbatim: "Now Playing",
                style: theme.typography.title,
                color: theme.colors.textPrimary
            )
            .padding(theme.spacing.lg)
        }
    }
}

File: android/core/ui/src/main/java/com/mplayeraudio/core/ui/preview/MultiplayerPreview.kt
package com.mplayeraudio.core.ui.preview

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxScope
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.mplayeraudio.core.ui.components.MultiplayerSurface
import com.mplayeraudio.core.ui.components.MultiplayerText
import com.mplayeraudio.core.ui.theme.MultiplayerDesignSystem
import com.mplayeraudio.core.ui.theme.MultiplayerTheme

@Composable
fun MultiplayerPreview(
    modifier: Modifier = Modifier,
    darkTheme: Boolean = false,
    contentAlignment: Alignment = Alignment.TopStart,
    content: @Composable BoxScope.() -> Unit,
) {
    MultiplayerDesignSystem(darkTheme = darkTheme) {
        MultiplayerSurface(
            modifier = modifier.fillMaxSize(),
            shape = RoundedCornerShape(0.dp),
            color = MultiplayerTheme.colors.background,
        ) {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(MultiplayerTheme.spacing.lg),
                contentAlignment = contentAlignment,
                content = content,
            )
        }
    }
}

@Preview(showBackground = true, name = "Multiplayer Preview Text")
@Composable
private fun MultiplayerPreviewTextPreview() {
    MultiplayerPreview(contentAlignment = Alignment.Center) {
        MultiplayerText(text = "MultiPlayer")
    }
}

@Preview(showBackground = true, name = "Multiplayer Preview Surface")
@Composable
private fun MultiplayerPreviewSurfacePreview() {
    MultiplayerPreview(contentAlignment = Alignment.Center) {
        MultiplayerSurface(
            shape = RoundedCornerShape(MultiplayerTheme.radius.large),
            tonalElevation = MultiplayerTheme.elevation.level2,
            shadowElevation = MultiplayerTheme.elevation.level1,
        ) {
            Box(modifier = Modifier.padding(MultiplayerTheme.spacing.lg)) {
                MultiplayerText(
                    text = "Now Playing",
                    style = MultiplayerTheme.typography.title,
                )
            }
        }
    }
}
